builder/proxmox/config.go

```diff
@@ -0,0 +1,196 @@
+package proxmox
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/hashicorp/packer/common"
+	"github.com/hashicorp/packer/common/bootcommand"
+	"github.com/hashicorp/packer/common/uuid"
+	"github.com/hashicorp/packer/helper/communicator"
+	"github.com/hashicorp/packer/helper/config"
+	"github.com/hashicorp/packer/packer"
+	"github.com/hashicorp/packer/template/interpolate"
+	"github.com/mitchellh/mapstructure"
+)
+
+type Config struct {
+	common.PackerConfig    `mapstructure:",squash"`
+	common.HTTPConfig      `mapstructure:",squash"`
+	bootcommand.BootConfig `mapstructure:",squash"`
+	RawBootKeyInterval     string              `mapstructure:"boot_key_interval"`
+	BootKeyInterval        time.Duration       ``
+	Comm                   communicator.Config `mapstructure:",squash"`
+
+	ProxmoxURLRaw      string `mapstructure:"proxmox_url"`
+	ProxmoxURL         *url.URL
+	SkipCertValidation bool   `mapstructure:"insecure_skip_tls_verify"`
+	Username           string `mapstructure:"username"`
+	Password           string `mapstructure:"password"`
+	Node               string `mapstructure:"node"`
+
+	VMName string `mapstructure:"vm_name"`
+	VMID   int    `mapstructure:"vm_id"`
+
+	Memory  int          `mapstructure:"memory"`
+	Cores   int          `mapstructure:"cores"`
+	Sockets int          `mapstructure:"sockets"`
+	OS      string       `mapstructure:"os"`
+	NICs    []nicConfig  `mapstructure:"network_adapters"`
+	Disks   []diskConfig `mapstructure:"disks"`
+	ISOFile string       `mapstructure:"iso_file"`
+
+	TemplateName        string `mapstructure:"template_name"`
+	TemplateDescription string `mapstructure:"template_description"`
+	UnmountISO          bool   `mapstructure:"unmount_iso"`
+
+	ctx interpolate.Context
+}
+
+type nicConfig struct {
+	Model      string `mapstructure:"model"`
+	MACAddress string `mapstructure:"mac_address"`
+	Bridge     string `mapstructure:"bridge"`
+	VLANTag    string `mapstructure:"vlan_tag"`
+}
+type diskConfig struct {
+	Type            string `mapstructure:"type"`
+	StoragePool     string `mapstructure:"storage_pool"`
+	StoragePoolType string `mapstructure:"storage_pool_type"`
+	Size            string `mapstructure:"size"`
```

Comment:
Sure, it does go under a `disks` key already though, so it would stutter a bit? Eg
```json
      "disks": [
        {
          "disk_size": "5G",
          "..."
```

Consistency might be more valuable, though?

Reply:
In this case, I think I prefer the consistency, even if it a bit of a stutter.